src/libraries/tests.proj

```diff
@@ -332,6 +332,52 @@
           Condition="'$(TestAssemblies)' == 'true' and
                      '$(Coverage)' == 'true'" />
 
+  <!-- Build Apple app bundles using AppBundleRoot -->
+  <UsingTask Condition="'$(UseAppBundleRootForBuildingTests)' == 'true'"
+             TaskName="XcodeCreateProject"
+             AssemblyFile="$(AppleAppBuilderTasksAssemblyPath)" />
+
+  <UsingTask Condition="'$(UseAppBundleRootForBuildingTests)' == 'true'"
+             TaskName="XcodeBuildApp"
+             AssemblyFile="$(AppleAppBuilderTasksAssemblyPath)" />
+
+  <Target Condition="'$(UseAppBundleRootForBuildingTests)' == 'true'"
+          Name="BuildAppleAppBundles"
+          AfterTargets="Build">
+
+    <ItemGroup>
+      <AppBundles Include="$(AppBundleRoot)/*/AppBundle/CMakeLists.txt" />
+      <CMakeEntries Include="project(AllAppBundles)" />
+      <CMakeEntries Include="add_subdirectory(%(AppBundles.RootDir)%(AppBundles.Directory) %(AppBundles.RecursiveDir) EXCLUDE_FROM_ALL)" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(AppBundleRoot)/all/CMakeLists.txt" Lines="@(CMakeEntries)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+
+    <XcodeCreateProject
+        TargetOS="$(TargetOS)"
+        Arch="$(TargetArchitecture)"
+        ProjectName="AllAppBundles"
+        CMakeListsDirectory="$(AppBundleRoot)/all" />
+
+    <ItemGroup>
+      <!-- xcodeproj are not files but directories -->
+      <XcodeProjects Include="$([System.IO.Directory]::GetDirectories('$(AppBundleRoot)/all/AllAppBundles/%(AppBundles.RecursiveDir)', '*.xcodeproj'))" />
+    </ItemGroup>
+
+    <MakeDir Directories="$(TestArchiveTestsDir)" />
+
+    <XcodeBuildApp
+        TargetOS="$(TargetOS)"
+        Arch="$(TargetArchitecture)"
+        XcodeProjectPath="%(XcodeProjects.Identity)"
+        DevTeamProvisioning="$(DevTeamProvisioning)"
+        Optimized="True"
+        DestinationFolder="$(TestArchiveTestsDir)" /> <!-- TODO: TestArchiveTestsRoot is different for when IsFunctionalTest==true but this is only set in the context of the .csproj so we need to find another solution here -->
```

Comment:
@steveisok we'll need to fix this once we turn on functional tests for iOS again.

Reply:
hmm actually we're still running functional tests on simulators right? I guess it'll fail there :)

Reply:
Yes, we're only skipping functional tests on devices for now since mlaunch can't handle the return code we expect.